### PR TITLE
Added required parameters to ISM Transforms Update

### PR DESCRIPTION
**Pull request by keithhc2**
#347
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by ashwinkumar12345 — APPROVED** [review]

File: _im-plugin/index-transforms/transforms-apis.md [hunk 1]
```diff
@@ -128,17 +128,17 @@ You can specify the following options in the HTTP request body:
...
 description | String | Describes the transform job. | No
 metadata_id | String | Any metadata to be associated with the transform job. | No
 source_index | String | The source index whose data to transform. | Yes
 target_index | String | The target index the newly transformed data is added into. You can create a new index or update an existing one. | Yes
-data_selection_query | JSON | The query DSL to use to filter a subset of the source index for the transform job. See [query DSL]({{site.url}}{{site.baseurl}}/opensearch/query-dsl) for more information. | Yes
-page_size | Integer | The number of fields to transform at a time. Higher number means higher performance but requires more memory and can cause higher latency. (Default: 1) | Yes
+data_selection_query | Object | The query DSL to use to filter a subset of the source index for the transform job. See [query DSL]({{site.url}}{{site.baseurl}}/opensearch/query-dsl) for more information. | Yes
+page_size | Integer |  The maximum number of buckets OpenSearch can use to transform data. Higher number means higher performance but requires more memory and can cause higher latency. | Yes
```
Comment (downsrob) [inline _im-plugin/index-transforms/transforms-apis.md:138]: This is a bit misleading as it seems to say that it limits the number of buckets which can be transformed, which isn't right. What actually occurs is that internally, in each transform job run, we transform all of the buckets, and the page size determines how many buckets we index at a time as we batch through the produced buckets.

File: _im-plugin/index-transforms/transforms-apis.md [hunk 2]
```diff
@@ -250,12 +250,23 @@ PUT _plugins/_transform/<transform_id>
...
+You can update the following fields:
+
+Option | Data Type | Description
+:--- | :--- | :---
+schedule | Object | The schedule the transform job runs on. Contains the fields `interval.start_time`, `interval.period`, and `interval.unit`.
+start_time | Integer | The Unix epoch start time of the transform job.
+period | Integer | How often to execute the transform job.
+unit | String | The unit of time associated with the execution period. Available options are `Minutes`, `Hours`, and `Days`.
+description | Integer | Describes the transform job.
+page_size | Integer | The number of fields to transform at a time. Higher number means higher performance but requires more memory and can cause higher latency.
```
Comment (downsrob) [inline _im-plugin/index-transforms/transforms-apis.md:269]: This page size description doesn't match the other

**Review by lizsnyder — APPROVED** [review]

File: _im-plugin/index-transforms/transforms-apis.md [hunk 3]
```diff
@@ -128,17 +128,17 @@ You can specify the following options in the HTTP request body:
...
 metadata_id | String | Any metadata to be associated with the transform job. | No
 source_index | String | The source index whose data to transform. | Yes
 target_index | String | The target index the newly transformed data is added into. You can create a new index or update an existing one. | Yes
-data_selection_query | JSON | The query DSL to use to filter a subset of the source index for the transform job. See [query DSL]({{site.url}}{{site.baseurl}}/opensearch/query-dsl) for more information. | Yes
-page_size | Integer | The number of fields to transform at a time. Higher number means higher performance but requires more memory and can cause higher latency. (Default: 1) | Yes
+data_selection_query | Object | The query DSL to use to filter a subset of the source index for the transform job. See [query DSL]({{site.url}}{{site.baseurl}}/opensearch/query-dsl) for more information. | Yes
+page_size | Integer | The number of buckets ISM processes and indexes concurrently. Higher number means better performance but requires more memory. If your machine runs out of memory, ISM automatically adjusts this field and retries until the operation succeeds. | Yes
```
Comment (thalurur) [inline _im-plugin/index-transforms/transforms-apis.md:138]: minor: ISM -> IM

**Review by thalurur — APPROVED** [review]

File: _opensearch/install/plugins.md [hunk 4]
```diff
@@ -54,38 +54,18 @@ bin/opensearch-plugin list
...
-analysis-ukrainian                   1.1.0
-discovery-azure-classic              1.1.0
-discovery-ec2                        1.1.0
-discovery-gce                        1.1.0
-ingest-attachment                    1.1.0
-mapper-annotated-text                1.1.0
-mapper-murmur3                       1.1.0
-mapper-size                          1.1.0
-opensearch-alerting                  1.2.0.0
+      <pre>opensearch-alerting                  1.2.0.0
```
Comment (Naarcha-AWS) [inline _opensearch/install/plugins.md:57]: Should this have a `-` like line 57?

**Review by Naarcha-AWS — APPROVED** [review]

File: _opensearch/install/plugins.md (hunk 4 above)
Comment (keithhc2) [reply _opensearch/install/plugins.md:57]: That's from GitHub, actually.

Comment (Naarcha-AWS) [reply _opensearch/install/plugins.md:57]: I realized that as I came back from lunch :D

**Pull request merged**